models/recipe.py

```diff
@@ -0,0 +1,143 @@
+        """ method that defines the elements required to create an account """
+        if recipe_name:
+            recipe_name = re.sub(r'\s+', ' ', recipe_name).strip()
+
+        recipe_name = None if recipe_name == " " else recipe_name.title()
+
+        if recipe_ingredients:
+            recipe_ingredients = re.sub(
+                r'\s+', ' ', recipe_ingredients).strip()
+
+        recipe_ingredients = None if recipe_ingredients == " " else recipe_ingredients
+
+        if recipe_methods:
+            recipe_methods = re.sub(
+                r'\s+', ' ', recipe_methods).strip()
+
+        recipe_methods = None if recipe_methods == " " else recipe_methods
+
+        if not (recipe_name):
+            return "Null recipe name"
+
+        if not (recipe_ingredients):
+            return "Null recipe ingredients"
+
+        if not (recipe_methods):
+            return "Null recipe method"
+
+        if not re.search(self.regex_name, recipe_name):
+            return "Recipe name has special characters"
+
+        if (any(recipe_name == recipe_list[1]
+                and owner == recipe_list[2]
+                and cat_name == recipe_list[0]
+                for recipe_list in self.recipes)) == True:
+            # returns an error if the recipe name is in the same category
+            # and having the same owner
+
+            return "Recipe exists"
+
+
+        # if the recipe name meets the condition it is added to the recipes list
+        self.recipes.append([cat_name, recipe_name, owner, recipe_ingredients, recipe_methods])
+        return "successfully created recipe"
+
+    def recipe_edit(self, new_recipe_name, recipe_name, cat_name, owner, recipe_ingredients, recipe_methods):
+        """ method that defines how to edit a recipe, the method takes 
+            the new recipe name as a parameter, the recipe name and the category
+            name, all belonging to the logged in user
+        """
+        if new_recipe_name:
+            new_recipe_name = re.sub(r'\s+', ' ', new_recipe_name).strip()
+
+        new_recipe_name = None if new_recipe_name == " " else new_recipe_name
+
+        if recipe_ingredients:
+            recipe_ingredients = re.sub(
+                r'\s+', ' ', recipe_ingredients).strip()
+
+        recipe_ingredients = None if recipe_ingredients == " " else recipe_ingredients
+
+        if recipe_methods:
+            recipe_methods = re.sub(
+                r'\s+', ' ', recipe_methods).strip()
+
+        recipe_methods = None if recipe_methods == " " else recipe_methods
+
+
+        if not (new_recipe_name):
+            return "Null recipe name"
+
+        if not re.search(self.regex_name, new_recipe_name):
+            return "Recipe name has special characters"
+
+        for recipe_list in self.recipes:
+
+            # checks to find if the new recipe name exists in the recipes list
+            if (any(new_recipe_name == recipe_list[1] for
+                    recipe_list in self.recipes)) == True:
```

Comment:
` == True` is not necessary. In Python, a condition is evaluated and a `Boolean` is returned. Eg:
`if x == False` is the same as writing `if not x`